### PR TITLE
Record JRE version details in Java image metadata

### PR DESCRIPTION
**Pull request by briandealwis**
#276 requests that the `java` images provide information as to the JRE version. This PR follows on from #290 and adds the following metadata to the `distroless/java` images:
  - it sets a `JAVA_VERSION` environment variable to the JRE version, currently `8u181` for the Java 8 image and `11.0.1` for the Java 11 image
  - ~~it sets three labels:~~
    - ~~`jdk.vendor` is hard-coded to `openjdk` since we explicitly select the `openjdk-*-jre-headless` packages~~
    - ~~`jdk.version` is the same release number reported in `JAVA_VERSION`~~
    - ~~`jdk.source` records soe provenance information listing the Debian package name and version: currently `openjdk-8-jre-headless 8u181-b13-2~deb9u1` and `openjdk-11-jre-headless 11.0.1+13-2~bpo9+1`~~

_Update: I'm not convinced of these label keys, and since environment variables are as easy to access as labels, the `JAVA_VERSION` will do for now._

It adds some metadata tests too ~~though the `jdk.source` tests are commented out as they require container-structure-test v1.7.0 (bazelbuild/rules_docker#685)~~.

~~I'm not particularly enamoured by the label names.~~  My priority is the `JAVA_VERSION` environment variable.

### CODE REVIEW

**Comment by chanseokoh** [comment]
There are formatting errors. You need to use the same `buildifier` version being used in Travis: `wget https://github.com/bazelbuild/buildtools/releases/download/0.11.1/buildifier`

Once #290 is merged, I expect the updated `dpkg_parser.par` will become available soon. You can then update the URL in `package_manager.bzl` to pull in the new binary, I think.

**Comment by briandealwis** [comment]
Ok will do — thanks for the pointer.

**Comment by loosebazooka** [comment]
Is the problem with the label keys that we're not choosing standard ones?

File: java/jre_ver.bzl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def jre_ver(version):
+  """Extract JRE version from an OpenJDK debian package.
+    debian packages have versions in the form:
+       openjdk-8-jre*: 8u171-b11-1~deb9u1
+       openjdk-11-jre*: 11.0.1+13-2~bpo9+1
+  """
+  if version.startswith("8u"): return version.split("-",2)[0]
+  if version.startswith("11."): return version.split("+",2)[0]
```
Comment (chanseokoh) [inline java/jre_ver.bzl]: nit: I believe `version.split("+")[0]` (no optional second parameter) will work and look simpler. I don't think we need this kind of optimization.

File: java/jre_ver.bzl [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  return "unknown: " + version
```
Comment (chanseokoh) [inline java/jre_ver.bzl]: Hmm... would it be difficult to not set `JAVA_VERSION` if unknown?

File: java/jre_ver.bzl (hunk 1 above)
Comment (loosebazooka) [reply java/jre_ver.bzl]: Agree. Perhaps the consumer can handle this on their own.

File: java/jre_ver.bzl (hunk 2 above)
Comment (briandealwis) [reply java/jre_ver.bzl]: I'd actually like to trigger a build error.  Thinking to the day that we pull in Java 15 and end up with a broken version string.

Comment (briandealwis) [reply java/jre_ver.bzl]: Ok, that should use the `fail()` function.

Comment (chanseokoh) [reply java/jre_ver.bzl]: Nice! I like that idea.

**Comment by chanseokoh** [comment]
I wonder why the status of the "push" Travis isn't updated. The build actually succeeded. I retriggered the build a few hours ago, but it didn't help.

File: java/jre_ver.bzl (hunk 2 above)
Comment (briandealwis) [reply java/jre_ver.bzl]: `fail()` does the trick:
```
ERROR: /Users/bsd/Manumitting/Projects/GooglePluginEclipse/repo-distroless/java/BUILD:13:1: Traceback (most recent call last):
	File "/Users/bsd/Manumitting/Projects/GooglePluginEclipse/repo-distroless/java/BUILD", line 13
		container_image(name = rule_name, base = "//cc" if...", <5 more arguments>)
	File "/Users/bsd/Manumitting/Projects/GooglePluginEclipse/repo-distroless/java/BUILD", line 27, in container_image
		jre_ver(versions[jre_deb])
	File "/Users/bsd/Manumitting/Projects/GooglePluginEclipse/repo-distroless/java/jre_ver.bzl", line 10, in jre_ver
		fail(("unrecognized openjdk package v...))
unrecognized openjdk package version: 15.0.2
ERROR: error loading package 'java': Package 'java' contains errors
INFO: Elapsed time: 0.093s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (1 packages loaded)
```

File: package_manager/dpkg_parser.py [hunk 3]
```diff
@@ -82,6 +82,7 @@ def download_dpkg(package_files, packages, workspace_name):
     """
     pkg_vals_to_package_file_and_sha256 = {}
     package_to_rule_map = {}
+    package_to_version_map = {}
```
Comment (chanseokoh) [inline package_manager/dpkg_parser.py]: I guess these changes will go away when you merge with master.

**Comment by briandealwis** [comment]
> @loosebazooka: Is the problem with the label keys that we're not choosing standard ones?

Yeah. Both [Docker](https://docs.docker.com/config/labels-custom-metadata/#key-format-recommendations) and [OCI](https://github.com/opencontainers/image-spec/blob/master/annotations.md) recommend namespacing labels/annotations using reverse-DNS prefixes.

**Pull request merged**